best-practices-catalog-and-collection.md

```diff
@@ -182,78 +182,78 @@ just meant to give users a sense of what types of values they could expect.
-
-So if you are writing a STAC client it is recommended to start with just supporting these two types of published catalogs. In 
-turn, if your data is published online publicly or for use on an intranet then following these recommendations will ensure
-that a wider range of clients will work with it.
+your STAC catalogs are easy to explore and resilient to any future changes to their layouts. There's three link-related
+decisions to make when building your STAC catalog:
+
+- Should your root catalog have a `self` link?
+- Should you use relative or absolute links for the rest of the catalog?
+- Should you use relative or absolute hrefs for assets?
+
+We discuss each of these decisions in turn.
+
+### Should your root catalog have a `self` link?
+
+If you're providing access to your catalog over the internet from a single fixed url, it is best practice to include a self
+link on the root catalog. In previous versions of these best practices, this was known as a "published" catalog.
+Note that `self` links [must be absolute](https://github.com/radiantearth/stac-spec/blob/master/commons/links.md#relation-types).
+
+Catalogs that are used locally or that are intended to be frequently copied should not have a `self` link on the root catalog.
+This makes them more portable, and in previous versions of these best practices this was known as a "self contained" catalog.
+The lack of a `self` link on the root catalog implies that all links and assets hrefs should be relative, as described in the
+next two sections.
+
+### Should you use relative or absolute links?
+
+As described above, if your root catalog does not have a `self` link, then all links and assets hrefs should be relative.
+
+If your root catalog does have a `self` link, then you should decide how often your catalog will be copied or moved. If the
+catalog's location is fixed at a single location, e.g. as a canonical metadata reference for a geospatial asset archive,
+then absolute links are appropriate. This enables downstream tools to more quickly traverse the catalog structure without
+having to resolve links relative to the root href. Additionally, if you are using absolute links for your entire catalog,
+every value in the catalog should have a `self` link.
+
+If your catalog is intended to be copied, if you anticipate changing its location, or if it might be served via
+different protocols (e.g. via a blob storage and via http) or through a proxy, then relative links should be used. This
+enables portability and usability regardless of the URL used to access the catalog. If the catalog is later moved, only
+a single link (the root catalog's `self` link) needs to be updated. Relative links imply that `self` links should
+_not_ be used, except possibly on the root catalog per the previous section.
+
+### Should you use relative or absolute hrefs for assets?
+
+The only case in which you should use relative hrefs for assets is if your assets are stored alongside your STAC values and
```

Comment:
STAC values?

Reply:
yeah that through me at first too. Should it be STAC objects?